src/group_theory/perm/sign.lean

```diff
@@ -761,3 +765,57 @@ end congr
 end sign
 
 end equiv.perm
+
+section alternating_subgroup
+open equiv.perm
+variables (α) [fintype α] [decidable_eq α]
+
+/-- The alternating group on a finite type, realized as a subgroup of `equiv.perm`.
+  For $A_n$, use `alternating_subgroup (fin n)`. -/
+def alternating_subgroup : subgroup (perm α) :=
+sign.ker
+
+variables {α}
+
+lemma alternating_subgroup_eq_sign_ker : alternating_subgroup α = sign.ker := rfl
+
+@[simp]
+lemma mem_alternating_subgroup {f : perm α} :
+  f ∈ alternating_subgroup α ↔ sign f = 1 :=
+monoid_hom.mem_ker _
+
+instance alternating_subgroup.fintype :
+  fintype (alternating_subgroup α) :=
```

Comment:
I'm surprised that you have to prove anything here - shouldn't typeclass inference find this automatically?

Reply:
Via `subtype.fintype`, as opposed to `fintype.subtype`

Reply:
`subtype.fintype` and `subgroup.fintype` each need some kind of decidability instance - I could, say, replace that this instance with
`instance : decidable_pred ((alternating_subgroup α).carrier) :=
λ f, decidable_of_iff (sign f = 1) mem_alternating_subgroup`.

I think I can probably make this work for the kernels of all `monoid_hom`s to `decidable_eq` monoids.

Reply:
I think `decidable_pred (\mem s)` is probably a better way to spell that, as it doesn't unfold `carrier` or set membership

Reply:
I added something similar for `monoid_hom.range` recently

Reply:
If I add this: ```
instance foo {G H : Type*} [group G] [group H] [decidable_eq G] [decidable_eq H]
  (f : G →* H) : decidable_pred (f.ker.carrier) :=
λ x, decidable_of_iff (f x = 1) f.mem_ker
```
then I get `fintype (sign.ker : subgroup (perm α))` but not `fintype (alternating_subgroup α)`.

Reply:
Yeah, you probably need to use `:= subtype.fintype` still to let it see through the definition. `@[derive fintype]` may or may not work too.

Reply:
Ok, I've found and pushed a smoother way to do this, hopefully it's smooth enough.

Reply:
Oh, `@[derive fintype]` is good.
I'm checking whether I still need that general decidability lemma.